src/allmydata/test/test_system.py

```diff
@@ -1266,7 +1295,9 @@ def _grab_stats(ignored):
             s = stats["stats"]
             self.failUnlessEqual(s["storage_server.accepting_immutable_shares"], 1)
             c = stats["counters"]
-            self.failUnless("storage_server.allocate" in c)
+            # Probably this should be Unicode eventually? But we haven't ported
+            # stats code yet.
+            self.failUnless(b"storage_server.allocate" in c)
```

Comment:
Maybe a link to the stats porting ticket here would be useful when time comes for that work?  Or a mention of this code in that ticket.  Or both.  I'm trying to think of what will save someone from spending 20 minutes trying to track down the cause of `test_upload_and_download_convergent` failing after they've ported the stats code and maybe made something unicode.  Maybe it won't be that bad since this change will be right inside an assertion...

Reply:
Opened ticket, linked it here.